integration_tests/integration/serviceProviderReferrals.spec.js

```diff
@@ -1826,300 +1908,344 @@ describe('Service provider referrals dashboard', () => {
-      cy.stubGetServiceUserByCRN(referral.referral.serviceUser.crn, deliusServiceUser)
-      cy.stubGetServiceCategory(serviceCategory.id, serviceCategory)
-      cy.stubGetAuthUserByUsername(serviceProvider.username, serviceProvider)
+      beforeEach(() => {
+        cy.stubGetSentReferralsForUserToken([])
+        cy.stubGetServiceProviderSentReferralsSummaryForUserToken([])
+        cy.stubGetIntervention(intervention.id, intervention)
+        cy.stubGetSupplierAssessment(referral.id, supplierAssessment)
+        cy.stubGetSentReferral(referral.id, referral)
+        cy.stubGetServiceUserByCRN(referral.referral.serviceUser.crn, deliusServiceUser)
+        cy.stubGetServiceCategory(serviceCategory.id, serviceCategory)
+        cy.stubGetUserByUsername(probationPractitioner.username, probationPractitioner)
+        cy.stubGetAuthUserByUsername(serviceProvider.username, serviceProvider)
 
-      cy.login()
+        cy.login()
+      })
 
-      cy.visit(`/service-provider/referrals/${referral.id}/progress`)
-      cy.get('#supplier-assessment-status').contains('not scheduled')
-      cy.contains('Schedule initial assessment').click()
-
-      cy.contains('Add appointment details')
-
-      cy.get('#date-day').type('24')
-      cy.get('#date-month').type('3')
-      cy.get('#date-year').type('2025')
-      cy.get('#time-hour').type('9')
-      cy.get('#time-minute').type('02')
-      cy.get('#time-part-of-day').select('AM')
-      cy.get('#duration-hours').type('1')
-      cy.get('#duration-minutes').type('15')
-      cy.contains('In-person meeting - Other locations').click()
-      cy.get('#method-other-location-address-line-1').type('Harmony Living Office, Room 4')
-      cy.get('#method-other-location-address-line-2').type('44 Bouverie Road')
-      cy.get('#method-other-location-address-town-or-city').type('Blackpool')
-      cy.get('#method-other-location-address-county').type('Lancashire')
-      cy.get('#method-other-location-address-postcode').type('SY4 0RE')
+      describe('with appointments in the past', () => {
+        describe('that happened earlier today', () => {
+          it('presents the feedback form as soon as the appointment is confirmed', () => {
+            const today = new Date()
+
+            cy.visit(`/service-provider/referrals/${referral.id}/progress`)
+            cy.get('#supplier-assessment-status').contains('not scheduled')
+            cy.contains('Schedule initial assessment').click()
+
+            // schedule page
+            cy.get('#date-day').type(today.getDate().toString())
+            cy.get('#date-month').type((today.getMonth() + 1).toString())
+            cy.get('#date-year').type(today.getFullYear().toString())
+            cy.get('#time-hour').type('0')
+            cy.get('#time-minute').type('0')
+            cy.get('#time-part-of-day').select('AM')
+            cy.get('#duration-hours').type('1')
+            cy.get('#duration-minutes').type('15')
```

Comment:
_not a blocker_ Random observation: this is quite a lot of repetition, does cypress support some kind of encapsulation for "blocks" like this?